Cabal/src/Distribution/Simple/Flag.hs

```diff
@@ -96,36 +101,47 @@ instance Enum a => Enum (Flag a) where
   enumFromThenTo (Flag a) (Flag b) (Flag c) = toFlag `map` enumFromThenTo a b c
   enumFromThenTo _ _ _ = []
 
+-- | Wraps a value in Flag.
 toFlag :: a -> Flag a
 toFlag = Flag
 
+-- | Extracts a value from a Flag a, and throws an exception on NoFlag.
+-- The Flag version of fromJust.
 fromFlag :: WithCallStack (Flag a -> a)
 fromFlag (Flag x) = x
 fromFlag NoFlag = error "fromFlag NoFlag. Use fromFlagOrDefault"
 
+-- | Extracts a value from a Flag a, and returns the default value on
+-- NoFlag. The Flag version of fromMaybe.
 fromFlagOrDefault :: a -> Flag a -> a
 fromFlagOrDefault _ (Flag x) = x
 fromFlagOrDefault def NoFlag = def
 
+-- | Converts a Flag a value to a Maybe a value.
 flagToMaybe :: Flag a -> Maybe a
 flagToMaybe (Flag x) = Just x
 flagToMaybe NoFlag = Nothing
 
--- | @since 3.4.0.0
+-- | Pushes a function through a Flag value, and returns a default
+-- if the Flag value is NoFlag. The Flag version of Data.Maybe.maybe.
```

Comment:
an eliminator (inductor/recursor) is an elminator, again I don't see pressing need to reference Maybe. We don't reference tuples for product types.